src/libmongoc/tests/test-mongoc-dns.c

```diff
@@ -497,6 +497,30 @@ test_srv_polling_mocked (void *unused)
    mongoc_uri_destroy (uri);
 }
 
+static void
+test_small_initial_buffer (void *unused)
+{
+   mongoc_rr_type_t rr_type = MONGOC_RR_SRV;
+   mongoc_rr_data_t rr_data;
+   bson_error_t error;
+   /* Size needs to be large enough to fit DNS answer header to not error, but
+    * smaller than SRV response to test. The SRV response is 155 bytes. This can
+    * be determined with: dig -t SRV _mongodb._tcp.test1.test.build.10gen.cc */
+   size_t small_buffer_size = 30;
+
+   memset (&rr_data, 0, sizeof (rr_data));
+   ASSERT_OR_PRINT (
+      _mongoc_client_get_rr ("_mongodb._tcp.test1.test.build.10gen.cc",
```

Comment:
Hard coding this is not fantastic, but it's a part of the initial seedlist discovery test suite and spec tests: https://github.com/mongodb/specifications/tree/master/source/initial-dns-seedlist-discovery/tests

If it does change in the future, there will necessarily be a spec change.